### PR TITLE
Writing-11168 Add workload for $dateTrunc expression in $group id calculation for time-series collection

### PR DESCRIPTION
**Pull request by lriuui0x0**

### CODE REVIEW

File: src/workloads/execution/TimeSeriesGroupDateTrunc.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  Timeseries performance test
```
Comment (jimoleary) [inline src/workloads/execution/TimeSeriesGroupDateTrunc.yml:4]: If you are going to merge, can we get a more complete description, including what the workload is for, what it does, what the phases are doing.

Also some keyword search terms  under 'Keywords' attribute.

File: src/workloads/execution/TimeSeriesGroupDateTrunc.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+Actors:
+- Name: Setup
+  Type: RunCommand
+  Thread: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Operations:
+      - OperationName: RunCommand
```
Comment (jimoleary) [inline src/workloads/execution/TimeSeriesGroupDateTrunc.yml:28]: Can you add a comment that the drop command is only here as a convenience to support running the test locally.  A sys-perf run would have an empty database.

**Review by jimoleary — CHANGES_REQUESTED** [review]
some minor documentation issues

**Review by jimoleary — DISMISSED** [review]
LGTM.

I created https://jira.mongodb.org/browse/TIG-4182 for the long branch name config, but I expect it will be a while before it is implemented,

**Review by jimoleary — COMMENTED** [review]
actually there are some linter errors.

**Comment by lriuui0x0** [comment]
evergreen merge

**Pull request merged**